orc8r/cloud/go/blobstore/ent_test.go

```diff
@@ -24,94 +25,208 @@ import (
 	magmaerrors "magma/orc8r/lib/go/errors"
 )
 
-func TestMigration(t *testing.T) {
+func TestSQLToEntMigration(t *testing.T) {
+	var tableName = "states"
 	db, err := sqorc.Open("sqlite3", ":memory:")
 	require.NoError(t, err)
-	fact := blobstore.NewSQLBlobStorageFactory("states", db, sqorc.GetSqlBuilder())
-	err = fact.InitializeFactory()
-	require.NoError(t, err)
-	storev1, err := fact.StartTransaction(nil)
+
+	sqlFact := blobstore.NewSQLBlobStorageFactory(tableName, db, sqorc.GetSqlBuilder())
+	err = sqlFact.InitializeFactory()
 	require.NoError(t, err)
-	blobs := blobstore.Blobs{
-		{Type: "type1", Key: "key1", Value: []byte("value")},
-		{Type: "type2", Key: "key2", Value: []byte("value")},
-		{Type: "type1", Key: "key2", Value: []byte("value")},
-	}
-	err = storev1.CreateOrUpdate("id1", blobs)
+
+	entFact := blobstore.NewEntStorage(tableName, db, nil)
+
+	checkBlobStoreMigrations(t, sqlFact, entFact)
+}
+
+func TestEntToSQLMigration(t *testing.T) {
+	var tableName = "states"
+	db, err := sqorc.Open("sqlite3", ":memory:")
 	require.NoError(t, err)
 
-	many, err := storev1.GetMany("id1", storage.TKs{
-		{Type: "type1", Key: "key1"},
-		{Type: "type2", Key: "key2"},
-	})
+	entFact := blobstore.NewEntStorage(tableName, db, nil)
+
+	sqlFact := blobstore.NewSQLBlobStorageFactory(tableName, db, sqorc.GetSqlBuilder())
+	err = sqlFact.InitializeFactory()
 	require.NoError(t, err)
-	require.Len(t, many, 2)
-	require.Equal(t, blobs[:2], many)
 
-	keys, err := storev1.GetExistingKeys([]string{"key1"}, blobstore.SearchFilter{})
+	checkBlobStoreMigrations(t, entFact, sqlFact)
+}
+
+func TestIntegration(t *testing.T) {
+	db, err := sqorc.Open("sqlite3", ":memory:")
 	require.NoError(t, err)
-	require.Equal(t, []string{"key1"}, keys)
+	fact := blobstore.NewEntStorage("states", db, sqorc.GetSqlBuilder())
+	integration(t, fact)
+}
+
+func checkBlobStoreMigrations(
+	t *testing.T,
+	fact1 blobstore.BlobStorageFactory,
+	fact2 blobstore.BlobStorageFactory,
+) {
+	var networkID = "id1"
+	var blobs = blobstore.Blobs{
+		blobstore.Blob{Type: "type1", Key: "key1", Value: []byte("value1")},
+		blobstore.Blob{Type: "type1", Key: "key2", Value: []byte("value2")},
+		blobstore.Blob{Type: "type2", Key: "key3", Value: []byte("value3")},
+		blobstore.Blob{Type: "type3", Key: "key3", Value: []byte("value4")},
+		blobstore.Blob{Type: "type4", Key: "key4", Value: []byte("value5")},
+	}
+	var blobNotInFact = blobstore.Blob{Type: "notSaved", Key: "notSavedKey", Value: []byte("notSavedValue")}
+
+	createBlobs(t, fact1, networkID, blobs)
+	checkReadBlobs(t, fact1, networkID, blobs, blobNotInFact)
+	checkReadBlobs(t, fact2, networkID, blobs, blobNotInFact)
+	checkWriteBlobs(t, fact2, networkID, blobs, blobNotInFact)
+}
 
-	err = storev1.Commit()
+// Assumes that expectedBlobs length is at least 2.
```

Comment:
Docstrings should be prefixed with the identifier -- so e.g.
```suggestion
// checkReadBlobs does xyz.
// Assumes that expectedBlobs length is at least 2.
```